R/outliers.R

```diff
@@ -179,7 +179,7 @@ detect_outlr_rm = function(x = seq_along(y), y, n = 21,
 
   # Calculate lower and upper thresholds and replacement value
   z = z %>%
-    epi_slide(fitted = median(y), n = n, align = "center") %>%
+    epi_slide(fitted = median(y), before = floor((n-1)/2), after = ceiling((n-1)/2)) %>%
```

Comment:
I think the documentation should explain what will happen with `n` here and who it relates to `before` and `after` in `epi_slide()`. Would it make more sense to add `before`/`after` as args to the function that default to these transformations?

Reply:
I asked Logan about this on Slack.

Reply:
And what was the answer? To be clear, I'm advocating either of the following:
1. In the documentation for `@param n` above, explain that if, for example, n=5, slide gets used under the hood with before=2 and after=2. Then link to `epi_slide()`.
2. Remove the n parameter and replace it with `before` / `after` directly.
If the decision was not to do 2, then you need to do 1.

Reply:
Oh, I see that you already did 1. Thanks!

Reply:
Debating on this.  Options:
- Explicate `@param n` in `detect_outlr_rm` and `@param n_threshold` in `detect_outlr_stl` that these are centered windows (+ maybe some more comments about what happens for even values).
- Change `n` -> `before`, `after` in `detect_outlr_rm`; just explain more about `@param n_threshold` in `detect_outlr_stl`.
- Change `n` and `n_threshold` to things involving `before` and `after`.

The later options add functionality and make naming more consistent with the updated slide, but also would introduce some naming mismatch in the stl one, which also has `n_trend`, `n_seasonal`.

Reply:
I'm happy with option 1. It's also the simplest.

Reply:
Logan directed me to this conversation in a message on slack. I think 1 (in Dan's list of options above) is an OK answer, but I see two possible arguments for doing 2 instead:

1. This would provide an interface that's more consistent with the other functions in the package that are being rewritten to use before/after
2. It allows for more-consistent behavior at the end of the time series where a centered window is not feasible (and in practice, at the end of the time series slider will truncate the window so that it would effectively have `before=2` and `after=0` in Dan's example with `n=5`). A user might be concerned about that and prefer to specify a window with `before=...` and `after=0` to get more stable results.

Reply:
Since Ken is trying to wrap up a bunch of PRs all at once this week, it probably makes sense just to do approach 1 + add an issue to revisit this later.  There is at least one other issue to consider regarding behavior at the end of the time series which I'd guess would be even more important: the fact that when data is trending, say, upward, the data at the end of the time series will be judged against recent data that is below it, VS. post-hoc we could see that it's in the middle of earlier data below it + later data above it.  Perhaps splitting `detect_outlr_rm`'s `n` -> `before`, `after` could be tackled at the same time, or maybe this will motivate some alternative approach that would be its own separate function.

Reply:
I'm not so sure now that there are other things to do as I claimed; using `detect_outlr_rm` with `before=k` and `after=0` would likely involve tweaking parameters so that they worked for historical trending-upward/downward regions, and so they should also be appropriate for the end of the time series.  Still, since Ken already implemented approach 1, I'll just file this in another issue.